go/libkb/device_clone_state.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+import (
+	"fmt"
+
+	jsonw "github.com/keybase/go-jsonw"
+)
+
+const (
+	DefaultCloneTokenValue string = "00000000000000000000000000000000"
+)
+
+type DeviceCloneState struct {
+	Prior  string
+	Stage  string
+	Clones int
+}
+
+type DeviceCloneStateJSONFile struct {
+	*JSONFile
+}
+type cloneDetectionResponse struct {
+	Status AppStatus `json:"status"`
+	Token  string    `json:"token"`
+	Clones int       `json:"clones"`
+}
+
+func (d *cloneDetectionResponse) GetAppStatus() *AppStatus {
+	return &d.Status
+}
+
+func UpdateDeviceCloneState(m *MetaContext) (before int, after int, err error) {
```

Comment:
Always pass `MetaContext` by value, and not by reference.

Reply:
Here and everywhere else. The point of `MetaContext` is to wrap `context.Context`, which you should always pass by value, and represents the call tree for the current "context", whether an RPC call from electron, or a background job, or whatever.  It also wraps `*GlobalContext`, which is shared across all threads of computation and represents global state.  So that you want to pass by reference.  `MetaContext`, if passed by value, satisfies both.

Reply:
ahh. interesting.